cg/apps/demultiplex/sample_sheet/create.py

```diff
@@ -26,19 +29,26 @@ def create_sample_sheet(
 
     flow_cell_sequencer: str = flow_cell.sequencer_type
 
-    if flow_cell_sequencer != Sequencers.NOVASEQ:
-        message = f"Only sample sheet generation for NovaSeq sequence data is currently supported. Found sequencer type: {flow_cell_sequencer}"
+    if flow_cell_sequencer == Sequencers.NOVASEQ:
```

Comment:
Here is a detailed description of the pattern and some examples if you want to use it: https://refactoring.guru/design-patterns/factory-method

I think it could make the code more readable :)

Reply:
Yeah, might be better in a separate PR. It would look something like this for example:

```python
def sample_sheet_creator_factory(sequencer, bcl_converter, flow_cell, lims_samples, run_parameters, force):
    if sequencer == Sequencers.NOVASEQ:
        return SampleSheetCreatorV1(bcl_converter, flow_cell, lims_samples, run_parameters, force)
    elif sequencer == Sequencers.NOVASEQX:
        return SampleSheetCreatorV2(bcl_converter, flow_cell, lims_samples, run_parameters, force)
    else:
        raise ValueError(f"Unsupported sequencer type: {sequencer}")

def create_sample_sheet(
    flow_cell: FlowCell,
    bcl_converter: BclConverter,
    lims_samples: List[FlowCellSample],
    run_parameters: RunParameters,
    force: bool = False,
) -> List[List[str]]:
    flow_cell_sequencer: str = flow_cell.sequencer_type
    try:
        sample_sheet_creator = sample_sheet_creator_factory(
            flow_cell_sequencer,
            bcl_converter,
            flow_cell,
            lims_samples,
            run_parameters,
            force,
        )
    except ValueError:
        message = f"Only demultiplexing of NovaSeq and NovaSeqX sequence data is currently supported. Found sequencer type: {flow_cell_sequencer}"
        LOG.warning(message)
        raise FlowCellError(message=message)

    LOG.info(
        f"Constructing a {bcl_converter} sample sheet for the {flow_cell_sequencer} flow cell {flow_cell.id}"
    )

    return sample_sheet_creator.construct_sample_sheet()

```